Sources/FoundationExtensions/Promise/Promise+HTTPStatusCode.swift

```diff
@@ -0,0 +1,16 @@
+import Combine
+import Foundation
+
+extension Publishers.Promise where Success == (data: Data, response: URLResponse), Failure == URLError {
+    public func validStatusCode() -> Publishers.Promise<Void, URLError> {
+        flatMap { _, response -> Publishers.Promise<Void, URLError> in
+            guard let httpResponse = response as? HTTPURLResponse else {
+                return Publishers.Promise(error: URLError(.badServerResponse))
+            }
+
+            return httpResponse.statusCode == 200
```

Comment:
200 is a bit tight for a general purpose library. There is at leasst 204 / 206 to consider and 3xx / redirects are also possible.

Reply:
Done